agent/embeddedconsul.go

```diff
@@ -0,0 +1,113 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"path/filepath"
+	"strconv"
+
+	consulAgent "github.com/hashicorp/consul/agent"
+	configAgent "github.com/hashicorp/consul/agent/config"
+	"github.com/hashicorp/consul/agent/connect"
+	"github.com/hashicorp/consul/sdk/freeport"
+	"github.com/hashicorp/go-uuid"
+)
+
+func NodeID() string {
+	id, err := uuid.GenerateUUID()
+	if err != nil {
+		panic(err)
+	}
+	return id
+}
+
+func randomPortsSource() (data string) {
```

Comment:
Don't pay too much attention to this and TestConfigHCL(); it was taken from consul's testagent.go as a quick hack to get it running. This will go.